landlab/components/river_bed_dynamics/river_bed_dynamics.py

```diff
@@ -0,0 +1,3433 @@
+        (id0, id1) = np.where(phi_i > 1.59)
+        if id0.shape[0] > 0:
+            G[id0, id1] = 5474 * (1 - 0.853 / phi_i[id0, id1]) ** 4.5
+
+        (id0, id1) = np.where((phi_i >= 1) & (phi_i <= 1.59))
+        if id0.shape[0] > 0:
+            G[id0, id1] = np.exp(
+                14.2 * (phi_i[id0, id1] - 1) - 9.28 * (phi_i[id0, id1] - 1) ** 2
+            )
+
+        (id0, id1) = np.where(phi_i < 1)
+        if id0.shape[0] > 0:
+            G[id0, id1] = phi_i[id0, id1] ** 14.2
+
+        Gf = grain_size_frequency * G
+        Gf_sum = np.sum(Gf, axis=1)
+
+        # Total bedload transport rate
+        W_star_s = 0.00218 * Gf_sum
+        self._sediment_transport__bedload_rate_link = (
+            ((np.sqrt(self._shear_stress_total / self._rho)) ** 3 * W_star_s)
+            / (self._R * self._g)
+        ) * np.sign(self._shear_stress)
+
+        # Now that bed load rate has been calculated in all links we replace those that
+        # were imposed. If there there are no imposed bedload GSD this will do nothing.
+        if self._fixed_link_calculate is True:
+            link_id_fixed_links = np.unique(self._fixed_surface_gsd_link_id)
+
+            self._sediment_transport__bedload_rate_link[
+                link_id_fixed_links
+            ] = self._sediment_transport__bedload_rate_imposed_link[link_id_fixed_links]
+
+        # During the calculation of fractional sediment transport, there might
+        # be instances where certain fractions will be zero. This situation
+        # could trigger a division by zero warning. However, we are aware that
+        # in these specific cases, there's no transport occurring. Thus, such a
+        # warning is not indicative of an actual problem in the calculation.
+        # To avoid unnecessary warnings, we've deactivated them.
+
+        np.seterr(divide="ignore", invalid="ignore")
+
+        # Frational bedload transport rate
+        p = np.zeros_like(Gf)
+        Gf_sum = np.transpose(np.tile(Gf_sum, (grain_size_frequency.shape[1], 1)))
+        p = Gf / Gf_sum
+        id0 = np.isnan(p)
+        p[id0] = 0
+        self._sediment_transport__bedload_grain_size_distribution_link = p
+
+        # Now that bedload GSD has been calculated in all links we replace those that
+        # were imposed. If there is no imposed bedload GSD this will do nothing.
+        if self._fixed_link_calculate:
+            linkId_0 = self._fixed_surface_gsd_link_id
+            linkId_1 = self._fixed_surface_gsd_link_fractions
+
+            self._sediment_transport__bedload_grain_size_distribution_link[
+                linkId_0, linkId_1
+            ] = self._sediment_transport__bedload_gsd_imposed_link[linkId_0, linkId_1]
+
+    def strainFunctions(self):
+        omega0 = np.zeros_like(self._phi_sgo)
+        sigma0 = np.zeros_like(self._phi_sgo)
+
+        # There are three intervals where we can interpolate Omega and Sigma
+        (I,) = np.where(self._phi_sgo <= 0.7639)
```

Comment:
Apart from being just a single letter, ``I`` isn't a great variable name to use as can look like a lowercase ``L`` in some editors.

Reply:
Thanks, done!